store/SPEC_STANDARD.md

```diff
@@ -0,0 +1,110 @@
+
+### Technical specification
+
+This is the main section of the document, and should contain protocol documentation, design rationale,
+required references, and technical details where appropriate. 
+Apart from the API sub-section which is required, the section may have any or all of the following sub-sections, as appropriate to the particular specification.
+
+- *API* - A detailed description of the features's API including which interfaces implements (for implementations) or extends (for interfaces).
+- *Technical Specification* - All technical details including syntax, diagrams, semantics, protocols, data structures, algorithms, and pseudocode as appropriate. The technical specification should be detailed enough such that separate correct implementations of the specification without knowledge of each other are compatible.
+- *Backwards Compatibility* - A discussion of compatibility (or lack thereof) with previous feature or protocol versions.
+- *Known Issues* - A list of known issues. This sub-section is specially important for specifications of already in-use features.
+- *Example Implementation* - A concrete example implementation or description of an expected implementation to serve as the primary reference for implementers. This sub-section is only expected to be included in newly proposed interfaces.
+
+### History
+
+A store specification should include a history section, listing any inspiring documents and a plaintext log of significant changes.
+
+See an example history section [below](#history-1).
+
+### Copyright
+
+A store specification should include a copyright section waiving rights via [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0).
+
+## Formatting
+
+### General
+
+Specifications must be written in GitHub-flavoured Markdown.
+
+For a GitHub-flavoured Markdown cheat sheet, see [here](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet). For a local Markdown renderer, see [here](https://github.com/joeyespo/grip).
+
+### Language
+
+Specifications should be written in Simple English, avoiding obscure terminology and unnecessary jargon. For excellent examples of Simple English, please see the [Simple English Wikipedia](https://simple.wikipedia.org/wiki/Main_Page).
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in specifications are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+### Pseudocode
+
+Pseudocode in specifications should be language-agnostic and formatted in a simple imperative standard, with line numbers, variables, simple conditional blocks, for loops, and
+English fragments where necessary to explain further functionality such as scheduling timeouts. LaTeX images should be avoided because they are difficult to review in diff form.
+
+Pseudocode for structs should be written in simple Typescript, as interfaces.
```

Comment:
should we use golang since that's the dominate language?

Reply:
I agree too, please suggest writing pseudo code for structs in Go with the Github flavour ```go

Reply:
I think that would be "dangerous". The goal of a specification should be to capture the essence of a design without many implementation details. If we allow golang, I am afraid that it would be too easy just to copy chunks of the code and paste them into the spec. Note that pieces of code, when relevant, may still have room in some subsection of the technical specification section, which is quite flexible. But I believe that the core of the spec should be language-agnostic. What do you think?

Also, it does not need to be typescript, this was just a proposal (that's what we use for [IBC specs](https://github.com/cosmos/ibc/tree/main/spec/core)).

Reply:
I agree with @angbrav, using golang makes it very easy to just copy-paste, when we actually want high-level, abstract pseudocode.

Ideally translation to/from the spec should require a tiny bit of effort, to make sure that the implementation is in line with the spec.

Reply:
the main issue here, Is some people will spend more time figuring out types in ts than writing the code.